### ISSUE TITLE
[a11y]: MultiSelect - Accesibility checker reports an issue with nested buttons for selected options

### ISSUE THREAD

**Issue by matas-at-ibm** [carbon-design-system/carbon#13345]
### Package

carbon-components-react

### Browser

Chrome, Firefox

### Operating System

MacOS

### Package version

v7.59.4

### React version

v17.0.2

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker - Latest Deployment

### Description

The list box component is a button, and the tag component (a descendant of the list box) that allows the clearance of selected options is also a button which causes the nested button issue(4.1.2)
![image](https://user-images.githubusercontent.com/25956901/225030405-b0ed74a2-f5b7-4972-b9a1-89d7c29d044d.png)

### WCAG 2.1 Violation

4.1.2

### Reproduction/example

https://stackblitz.com/edit/github-verpba?file=src/App.jsx

### Steps to reproduce

1. Visit carbon dropdown usage examples:
   - in the [official documentation](https://carbondesignsystem.com/components/dropdown/usage)
   - or in [storybook](https://react.carbondesignsystem.com/?path=/story/components-multiselect--with-initial-selected-items)
3. Select `MultiSelect` variant in the example
4. Select a couple of options inside the multi-select input
6. Run the **IBM Equal Access Accessibility Checker** scan

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tw15egan** [comment]
This is resolved with https://github.com/carbon-design-system/carbon/pull/13173, which was merged recently. It will be included in `@carbon/react@1.25.0` which will be released in the next week
